pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -27,7 +27,7 @@ assert svnSupport -> perlSupport;
 let
   version = "2.35.1";
   svn = subversionClient.override { perlBindings = perlSupport; };
-
+  isFull = svnSupport && guiSupport && sendEmailSupport;
```

Comment:
Can we have a separate value here? This seems like a hack. Like why not `pythonSupport`?

I think the best option is to use the presence of the openssh package as the flag.

Reply:
> I think the best option is to use the presence of the openssh package as the flag.

I like that idea :) For `git` it can default to `null` and then `gitFull` can override it to the actual package/default (`openssh`).

Reply:
Overwriting packages to null is an anti pattern which I am not going to add more off.
I can also the flags from the default and minimal git to count as full or a new flag. Either ways are better.

Reply:
If you don't like the null package pattern then a `withSsh`  sounds good.

Reply:
> Overwriting packages to null is an anti pattern

This is a personal opinion, unless it's documented somewhere (with consensus).
It's not ideal but so is the whole package customization situation.

> If you don't like the null package pattern then a `withSsh` sounds good.

Also sounds good to me.